apps/agent/README.md

```diff
@@ -0,0 +1,254 @@
+
+If the sender has the right ACL permissions, it will execute an EVM call to `_target`, sending the specified ETH amount (in Wei) and `_data` as the calldata.
+
+- If the call reverts, it will revert forwarding the error data as the error data of the main call frame.
+- If the call succeeds, it will emit the `Execute` event logging the arguments the function was called with.
+
+Since any data can be passed as argument, it is important to note that the function could be a possible attack vector. Consider using `safeExecute` for all calls that don't require the transfer of funds. See the [Security](#14-security) section for further details.
+
+### 1.2. Signature handling
+
+
+
+### 1.3. Vault
+
+The Agent app is a superset of the [Vault](../vault/contracts/Vault.sol) contract, which means that it can store or transfer the tokens and Ether of the DAO in which it is installed. Specifically, the following functions are available: 
+
+ - `deposit(address _token, uint256 _value)` 
+ - `transfer(address _token, address _to, uint256 _value)` - Protected by the `TRANSFER_ROLE` permission.
+ - `balance(address _token)`
+
+More importantly, this property means that Agent is capable of using the DAO's funds to interact seemlessly with any contracts that require the sender to possess Ether or ERC-20 tokens (e.g. Exchanging tokens on [Uniswap](https://uniswap.io), participating in a [PoolTogether](https://www.pooltogether.com) lottery).
+
+### 1.4. Security
+
+Since the Agent app can manage tokens and Ether, it is crucial to always keep security in mind. In order to restrict unauthorized transfers and other abusive behaviors, a variant of the `execute` function is available that will prevent any attempt to transfer tokens included in a "protected tokens" list:
+
+- `safeExecute(address _target, bytes _data)` - Protected by the `SAFE_EXECUTE_ROLE` permission.
+
+Before executing the requested call, the function fetches in memory every balances from the protected tokens and compares them with the new balances after the call is executed. If a balance has changed, it reverts the call.
+
+The following functions can be used to add or remove a token address to the protected tokens list:
+
+- `addProtectedToken(address _token)` - Protected by the `ADD_PROTECTED_TOKEN_ROLE` permission.
+- `removeProtectedToken(address _token)` - Protected by the `REMOVE_PROTECTED_TOKEN_ROLE` permission.
+
+### 1.5. Forwarder
+
+
+## 2. Entry points
+
+### 2.1. Initialize
+ - **Actor:** External entity, ultimately calling the function through the `Kernel`
+ - **Inputs:** None
+ - **Pre-flight checks:**
+   - Ensure the app has not already been initialized
+ - **State transitions:**
+   - Initialize the app
+
+### 2.2. Execute
+ - **Actor:** External entity
+ - **Inputs:** 
+   - **Target:** Address where the action is being executed
+   - **ETH value:** Amount of ETH (in Wei) sent with the action
+   - **Data:** Calldata for the action
+ - **Pre-flight checks:**
```

Comment:
I don't remember exactly, but there was some discussion around making sure `msg.data > 0` because otherwise you'd just use `.transfer()` to transfer ETH. We may want to highlight that behaviour.

Reply:
hmm.. I'm not sur I understand what you mean 🤔 Making sure `msg.data > 0` or making sure `_data > 0`?

Reply:
This was the change: https://github.com/aragon/aragon-apps/pull/625

And the currently deployed contract: https://github.com/aragon/aragon-apps/blob/master/apps/agent/contracts/Agent.sol#L71

I remembered we had a discussion about whether to enforce `execute` to have data, but we removed it in the end :).